solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/JCSMPSessionProducerManager.java

```diff
@@ -53,11 +54,15 @@ public void responseReceivedEx(Object correlationKey) {
 				ErrorQueueRepublishCorrelationKey key = (ErrorQueueRepublishCorrelationKey) correlationKey;
 				try {
 					key.handleSuccess();
-				} catch (SolaceStaleMessageException e) { // unlikely to happen
-					logger.warn(String.format("Message %s successfully sent to error queue %s, " +
-									"but the reference is now stale. Message is likely duplicated and was/will be" +
-									" redelivered on the original queue.",
-							key.getSourceMessageId(), key.getErrorQueueName()), e);
+				} catch (SolaceAcknowledgmentException e) { // unlikely to happen
+					if (ExceptionUtils.indexOfType(e, IllegalStateException.class) > -1) {
+						logger.warn(String.format("Message %s successfully sent to error queue %s, " +
+										"but the reference is now stale. Message is likely duplicated and was/will be" +
+										" redelivered on the original queue.",
+								key.getSourceMessageId(), key.getErrorQueueName()), e);
+					} else {
```

Comment:
`IllegalStateException` isn't guaranteed to be thrown as a result of a stale message, right? Maybe it'd be better to just rethrow this?